cmd/anonymizer/main.go

```diff
@@ -69,9 +70,15 @@ func main() {
 			}
 
 			for _, span := range spans {
-				writer.WriteSpan(&span)
+				err := writerObj.WriteSpan(&span)
+				if err != nil {
+					logger.Fatal("error while writing span", zap.Error(err))
```

Comment:
it's not fatal if it's `ErrMaxSpansCountReached`. Also, Fatal will exit so you won't ever reach the next line.
* move logging after `if error.Is` check
* use `logger.Error`, not `Fatal`

Reply:
Made the changes.